src/registries/npm-registry.js

```diff
@@ -208,26 +208,33 @@ export default class NpmRegistry extends Registry {
       return this.token;
     }
 
-    const registry = this.getRegistry(packageName);
+    const baseRegistry = this.getRegistry(packageName);
+    const registries = [baseRegistry];
 
-    // Check for bearer token.
-    const authToken = this.getRegistryOrGlobalOption(registry, '_authToken');
-    if (authToken) {
-      return `Bearer ${String(authToken)}`;
+    if (baseRegistry === `https://registry.yarnpkg.com/`) {
+      registries.push(`https://registry.npmjs.org/`);
     }
 
-    // Check for basic auth token.
-    const auth = this.getRegistryOrGlobalOption(registry, '_auth');
-    if (auth) {
-      return `Basic ${String(auth)}`;
-    }
+    for (const registry of registries) {
```

Comment:
Is this trying to get an auth token for the first matching registry? If so, I find that a bit dangerous. Should we keep a mapping of `registry: token` pairs and use the appropriate one when communicating. This would also leak tokens to other registries which may be an important security threat.